packages/core/admin/admin/src/pages/SettingsPage/pages/ApiTokens/EditView/components/Regenerate/index.js

```diff
@@ -0,0 +1,86 @@
+import React, { useState } from 'react';
+import styled from 'styled-components';
+import PropTypes from 'prop-types';
+import { useIntl } from 'react-intl';
+import { Button } from '@strapi/design-system/Button';
+import Refresh from '@strapi/icons/Refresh';
+import { ConfirmDialog } from '@strapi/helper-plugin';
+import { axiosInstance } from '../../../../../../../core/utils';
+
+const ButtonWithRightMargin = styled(Button)`
+  margin-right: ${({ theme }) => theme.spaces[2]};
+`;
+
+export const Regenerate = ({ onRegenerate, idToRegenerate }) => {
+  let isLoadingConfirmation = false;
+  const { formatMessage } = useIntl();
+  const [showConfirmDialog, setShowConfirmDialog] = useState(false);
+  const handleConfirmRegeneration = async () => {
+    isLoadingConfirmation = true;
```

Comment:
Why did do decide to make this a non-reactive variable? I'm wondering if the loading state `isConfirmButtonLoading={isLoadingConfirmation}` is ever applied that way ...

Reply:
yes I forget to transform it to state

Reply:
done